src/keymap/keymap.rs

```diff
@@ -21,6 +21,19 @@ use crate::state::State;
 pub struct Keymap(*mut xkb_keymap);
 
 impl Keymap {
+	pub fn from_str(s: &str) -> Option<Self> {
+		unsafe{
+			let context = xkb_context_new(0);
```

Comment:
Where is the context freed? Seems like this is going to leak it.

Reply:
Corrected it now.